staging/src/k8s.io/apiserver/pkg/endpoints/handlers/update.go

```diff
@@ -101,6 +102,30 @@ func UpdateResource(r rest.Updater, scope RequestScope, admit admission.Interfac
 				return newObj, mutatingAdmission.Admit(admission.NewAttributesRecord(newObj, oldObj, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Update, userInfo))
 			})
 		}
+		transformers = append(transformers, func(ctx context.Context, newObj, oldObj runtime.Object) (runtime.Object, error) {
+			if hasUID, err := hasUID(oldObj); err != nil {
+				return nil, err
+			} else if !hasUID && scope.Authorizer != nil {
+				createAttributes := authorizer.AttributesRecord{
```

Comment:
we should only need to run this check once per request... hoist out into a `do.Once`?

Reply:
It shouldn't be harmful to call multiple times, other than it takes time. But, I think if we want to cache decisions, we should do that in whatever is implementing the Authorize method, rather than ad-hoc by only calling once per request. That will make it easier to provide guarantees on how long things are cached and what the propagation delay therefore might be for a rule change.

Reply:
> It shouldn't be harmful to call multiple times, other than it takes time. But, I think if we want to cache decisions, we should do that in whatever is implementing the Authorize method, rather than ad-hoc by only calling once per request.

Thinking of authorization as a logical layer, once you're through the layer for a particular API operation, the same request doesn't go back through again. The only reason we're checking here is that you haven't yet made it through the layer for the create operation. Especially in cases like patch where the server retries application of the object as part of persisting a single request, I don't think we want to recall this many times.